packages/x-date-pickers/src/ClockPicker/ClockPicker.test.tsx

```diff
@@ -170,6 +170,28 @@ describe('<ClockPicker />', () => {
     expect(reason).to.equal('partial');
   });
 
+  it('should call `shouldDisableTime` with the hours with meridiem', () => {
+    const shouldDisableTime = spy(() => false);
+
+    render(
+      <ClockPicker
+        autoFocus
+        date={adapterToUse.date('2019-01-01T18:20:00.000')}
+        onChange={() => {}}
+        shouldDisableTime={shouldDisableTime}
+        ampm
+      />,
+    );
+
+    const hours = shouldDisableTime
+      .getCalls()
+      .filter((el) => el.lastArg === 'hours')
+      .map((el) => el.firstArg);
+
+    // Should be called with every hour from 0PM to 11PM (12 - 23) + with the current date hour
+    expect(Math.max(...hours)).to.equal(23);
```

Comment:
I get a bit confused about what you are testing
```suggestion
    // Should be called with every hour post meridiem (from 12 to 23) since current date hour is 6PM
    expect(Math.min(...hours)).to.equal(12);
    expect(Math.max(...hours)).to.equal(23);
```